### PR TITLE
You can now join in as a buried larva without a pool at roundstart

### PR DESCRIPTION
**Pull request by carlarctg**
# About the pull request
Larvae can now pop out from the ground nearby the hive core, the queen, builder castes, larvae, or any xeno in that order, during the first 20 minutes of the round.

Changed the span of a queen announcement.

# Explain why it's good for the game

> Larvae can now pop out from the ground nearby the hive core, the queen, builder castes, larvae, or any xeno in that order, during the first 20 minutes of the round.

This is intended to let people quickly join in as xeno even if the pool wasn't built, preventing things like forgetting to build it from absolutely crippling the hive early on.

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:
add: Larvae can now pop out from the ground nearby the hive core, the queen, builder castes, larvae, or any xeno in that order, during the first 20 minutes of the round.
/:cl:

### CODE REVIEW

File: code/game/gamemodes/cm_initialize.dm [hunk 1]
```diff
@@ -369,30 +368,35 @@ Additional game mode variables.
...
 						return
 				if(isnewplayer(xeno_candidate))
-					var/mob/new_player/N = xeno_candidate
-					N.close_spawn_windows()
-				SP.spawn_pooled_larva(xeno_candidate)
+					var/mob/new_player/noob = xeno_candidate
+					noob.close_spawn_windows()
+				if(picked_hive.spawn_pool)
+					picked_hive.spawn_pool.spawn_pooled_larva(xeno_candidate)
+				else if((world.time < 30 MINUTES + SSticker.round_start_time))
```
Comment (stanalbatross) [inline code/game/gamemodes/cm_initialize.dm:392]: make this 30 minutes value a define and drop to 20

File: code/game/gamemodes/cm_initialize.dm [hunk 2]
```diff
@@ -369,30 +368,35 @@ Additional game mode variables.
...
-						to_chat(xeno_candidate, SPAN_WARNING("You must wait 2.5 minutes before rejoining the game!"))
+						to_chat(xeno_candidate, SPAN_WARNING("You must wait 2.5 minutes before rejoining the game as a buried larva!"))
 						return FALSE
 
-				for(var/mob_name in SP.linked_hive.banished_ckeys)
-					if(SP.linked_hive.banished_ckeys[mob_name] == xeno_candidate.ckey)
-						to_chat(xeno_candidate, SPAN_WARNING("You are banished from this hive, You may not rejoin unless the Queen re-admits you or dies."))
+				for(var/mob_name in picked_hive.banished_ckeys)
+					if(picked_hive.banished_ckeys[mob_name] == xeno_candidate.ckey)
+						to_chat(xeno_candidate, SPAN_WARNING("You are banished from this picked_hive, You may not rejoin unless the Queen re-admits you or dies."))
```
Comment (stanalbatross) [inline code/game/gamemodes/cm_initialize.dm:385]:
```suggestion
						to_chat(xeno_candidate, SPAN_WARNING("You are banished from the [picked_hive], You may not rejoin unless the Queen re-admits you or dies."))
```
did you mean this?

File: code/game/gamemodes/cm_initialize.dm [hunk 3]
```diff
@@ -369,30 +368,35 @@ Additional game mode variables.
...
+			if(picked_hive.stored_larva)
 				if(!xeno_bypass_timer)
 					var/deathtime = world.time - xeno_candidate.timeofdeath
 					if(isnewplayer(xeno_candidate))
 						deathtime = 2.5 MINUTES //so new players don't have to wait to latejoin as xeno in the round's first 5 mins.
 					if(deathtime < 2.5 MINUTES && !check_client_rights(xeno_candidate.client, R_ADMIN, FALSE))
 						var/message = SPAN_WARNING("You have been dead for [DisplayTimeText(deathtime)].")
 						to_chat(xeno_candidate, message)
-						to_chat(xeno_candidate, SPAN_WARNING("You must wait 2.5 minutes before rejoining the game!"))
+						to_chat(xeno_candidate, SPAN_WARNING("You must wait 2.5 minutes before rejoining the game as a buried larva!"))
```
Comment (stanalbatross) [inline code/game/gamemodes/cm_initialize.dm:380]: what if you made this actually calculate how long they had left :3

**Review by stanalbatross — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/cm_initialize.dm (hunk 3 above)
Comment (carlarctg) [reply code/game/gamemodes/cm_initialize.dm:380]: i don't think that's worth it, they can already infer it from the var/message, best to actually show the total duration

File: code/game/gamemodes/cm_initialize.dm (hunk 1 above)
Comment (carlarctg) [reply code/game/gamemodes/cm_initialize.dm:392]: don

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm [hunk 4]
```diff
@@ -800,6 +800,41 @@
...
+	if(!SSticker.mode.transfer_xeno(xeno_candidate, new_xeno))
+		qdel(new_xeno)
+		return FALSE
+	new_xeno.visible_message(SPAN_XENODANGER("A larva suddenly burrows out of \the [spawning_turf]!"),
+	SPAN_XENODANGER("You burrow out of \the [spawning_turf] and awaken from your slumber. For the Hive!"))
+	msg_admin_niche("[key_name(new_xeno)] burrowed out from \a [spawning_turf]. (<A HREF='?_src_=admin_holder;[HrefToken(forceGlobal = TRUE)];adminplayerobservecoodjump=1;X=[spawning_turf.x];Y=[spawning_turf.y];Z=[spawning_turf.z]'>JMP</a>)")
+	playsound(new_xeno, 'sound/effects/xeno_newlarva.ogg', 50, 1)
+	to_chat(new_xeno, SPAN_XENOANNOUNCE("You are a xenomorph larva awakened from slumber!"))
+	if(new_xeno.client)
+		if(new_xeno.client?.prefs.toggles_flashing & FLASH_POOLSPAWN)
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:832]:
```suggestion
		if(new_xeno.client?.prefs?.toggles_flashing & FLASH_POOLSPAWN)
```

**Review by stanalbatross — APPROVED** [review]

File: code/game/gamemodes/cm_initialize.dm [hunk 5]
```diff
@@ -369,30 +368,35 @@ Additional game mode variables.
...
 						return FALSE
 
-				for(var/mob_name in SP.linked_hive.banished_ckeys)
-					if(SP.linked_hive.banished_ckeys[mob_name] == xeno_candidate.ckey)
-						to_chat(xeno_candidate, SPAN_WARNING("You are banished from this hive, You may not rejoin unless the Queen re-admits you or dies."))
+				for(var/mob_name in picked_hive.banished_ckeys)
+					if(picked_hive.banished_ckeys[mob_name] == xeno_candidate.ckey)
+						to_chat(xeno_candidate, SPAN_WARNING("You are banished from the [picked_hive], You may not rejoin unless the Queen re-admits you or dies."))
```
Comment (harryob) [inline code/game/gamemodes/cm_initialize.dm:385]:
```suggestion
						to_chat(xeno_candidate, SPAN_WARNING("You are banished from the [picked_hive], you may not rejoin unless the Queen re-admits you or dies."))
```

**Review by harryob — CHANGES_REQUESTED** [review]

**Review by harryob — APPROVED** [review]

**Pull request merged**